graphql/src/store/query.rs

```diff
@@ -161,91 +161,144 @@ fn parse_change_block_filter(value: &r::Value) -> Result<BlockNumber, QueryExecu
-                        field_name.clone(),
-                    )
-                })?;
+            .map(|item| {
+                return match item {
+                    r::Value::Object(object) => {
+                        Ok(build_filter_from_object(entity, object, schema)?)
+                    }
+                    _ => Err(QueryExecutionError::InvalidFilterError),
+                };
+            })
+            .collect::<Result<Vec<Vec<EntityFilter>>, QueryExecutionError>>()?
+            // Flatten all different EntityFilters into one list
+            .into_iter()
+            .flatten()
+            .collect::<Vec<EntityFilter>>()),
+        _ => Err(QueryExecutionError::InvalidFilterError),
+    };
+}
 
-                let ty = &field.field_type;
+/// build a filter which has list of nested filters
+fn build_list_filter_from_object(
+    entity: ObjectOrInterface,
+    object: &Object,
+    schema: &ApiSchema,
+) -> Result<Vec<EntityFilter>, QueryExecutionError> {
+    Ok(object
+        .iter()
+        .map(|(_, value)| {
+            return build_list_filter_from_value(entity, schema, value);
+        })
+        .collect::<Result<Vec<Vec<EntityFilter>>, QueryExecutionError>>()?
+        .into_iter()
+        // We iterate an object so all entity filters are flattened into one list
+        .flatten()
+        .collect::<Vec<EntityFilter>>())
+}
 
-                Ok(match op {
-                    Child => match value {
-                        DataValue::Object(obj) => {
-                            build_child_filter_from_object(entity, field_name, obj, schema)?
-                        }
-                        _ => {
-                            return Err(QueryExecutionError::AttributeTypeError(
-                                value.to_string(),
-                                ty.to_string(),
-                            ))
-                        }
-                    },
+/// Parses a GraphQL input object into an EntityFilter, if present.
+fn build_filter_from_object(
+    entity: ObjectOrInterface,
+    object: &Object,
+    schema: &ApiSchema,
+) -> Result<Vec<EntityFilter>, QueryExecutionError> {
+    Ok(object
+        .iter()
+        .map(|(key, value)| {
+            // Special handling for _change_block input filter since its not a
+            // standard entity filter that is based on entity structure/fields
+            if key == "_change_block" {
+                return match parse_change_block_filter(value) {
+                    Ok(block_number) => Ok(EntityFilter::ChangeBlockGte(block_number)),
+                    Err(e) => Err(e),
+                };
+            }
+            use self::sast::FilterOp::*;
+            let (field_name, op) = sast::parse_field_as_filter(key);
+
+            Ok(match op {
+                And => EntityFilter::And(build_list_filter_from_object(entity, object, schema)?),
+                Or => EntityFilter::Or(build_list_filter_from_object(entity, object, schema)?),
```

Comment:
These two branches should check whether boolean filters are disabled and return a `QueryExecutionError` if they are.